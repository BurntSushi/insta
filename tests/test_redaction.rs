use insta::_macro_support::Selector;
use insta::{assert_debug_snapshot_matches, assert_serialized_snapshot_matches};
use serde::Serialize;
use uuid::Uuid;

#[test]
fn test_selector_parser() {
    macro_rules! assert_selector {
        ($short:expr, $sel:expr) => {
            assert_debug_snapshot_matches!($short, Selector::parse($sel).unwrap());
        };
    }

    assert_selector!("foo_bar", ".foo.bar");
    assert_selector!("foo_bar_alt", ".foo[\"bar\"]");
    assert_selector!("foo_bar_full_range", ".foo.bar[]");
    assert_selector!("foo_bar_range_to", ".foo.bar[:10]");
    assert_selector!("foo_bar_range_from", ".foo.bar[10:]");
    assert_selector!("foo_bar_range", ".foo.bar[10:20]");
}

#[test]
fn test_with_random_value() {
    #[derive(Serialize)]
    pub struct User {
        id: Uuid,
        username: String,
    }

    assert_serialized_snapshot_matches!("user", &User {
        id: Uuid::new_v4(),
        username: "john_doe".to_string(),
    }, {
        ".id" => "[uuid]"
    });
}
