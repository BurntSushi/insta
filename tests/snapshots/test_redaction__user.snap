---
created: "2019-01-21T22:03:13.792906+00:00"
creator: insta@0.3.0
expression: "&User{id: Uuid::new_v4(), username: \"john_doe\".to_string(),}"
source: tests/test_redaction.rs
---
id: "[uuid]"
username: john_doe
